Widgets/AssetView.cpp

```diff
@@ -0,0 +1,34 @@
+#include "AssetView.h"
+
+AssetView::AssetView(QWidget* parent) : QWidget(parent), zoom(1) {}
+
+void AssetView::SetZoom(qreal zoom) {
+  if (zoom > 3200) zoom = 3200;
+  if (zoom < 0.0625) zoom = 0.0625;
+  this->zoom = zoom;
+  setFixedSize(sizeHint() * zoom);
+}
+
+const qreal& AssetView::GetZoom() const { return zoom; }
+
+void AssetView::paintGrid(QPainter& painter, int width, int height, int gridHorSpacing, int gridVertSpacing,
+                          int gridHorOff, int gridVertOff, int gridWidth, int gridHeight) {
+  painter.setCompositionMode(QPainter::RasterOp_SourceXorDestination);
+  painter.setPen(QColor(0xff, 0xff, 0xff));
+
+  if (gridHorSpacing != 0 || gridVertSpacing != 0) {

```

Comment:
`&&`, or else handle when when or the other is 0

Reply:
I misread your variables; you aren't handling the case where `gridWidth` / `gridHeight` are zero. Is that handled elsewhere?

Reply:
Ok: a1efab0188be39b1a8a2ffe9eec39a0b954a1615

So fundies did it the way he did it because of half-pixel issues with QPainter/QPen. It draws the line off by a half a pixel. Regardless, fundies way is an inefficient way of dealing with that so I did change the logic the way you wanted.

Reply:
LGTM—another option to consider is just not drawing grid lines with spacing less than some predetermined constant (eg, 3), which would handle the 0 case naturally.